### PR TITLE
Implement prefix-lists in the generic routing module

### PR DESCRIPTION
**Pull request by ipspace**

### CODE REVIEW

**Comment by ipspace** [comment]
@ssasso -- just in case you'd have some time to review this. The next step will be 'match' conditions in routing policies, I'll probably do them in a separate PR

**Review by ssasso — APPROVED** [review]
Looks good! as soon as it enters in `dev` I will add the support for vyos and arubacx

File: netsim/ansible/templates/routing/eos.j2 [hunk 1]
```diff
@@ -8,3 +9,26 @@
 {{     routemap.common_route_map_entry(p_entry) }}
 {%-   endcall %}
 {% endif %}
+{% if routing._prefix.ipv4|default(False) %}
+!
+! IPv4 prefix lists
+!
+{%   for p_name,p_value in routing._prefix.ipv4.items() %}
```
Comment (ssasso) [inline netsim/ansible/templates/routing/eos.j2:16]: Isn't this a duplication, if the for loop is already performed by the included file `_prefix_list_ios.j2`?

(same for ipv6?)

Comment (ipspace) [reply netsim/ansible/templates/routing/eos.j2:16]: EOS has a different syntax for IPv6 prefix lists than for IPv4 ones, and once I figured that out, I decided to inline them both for consistency (and who knows what else might pop up later)

**Comment by ipspace** [comment]
> Looks good! as soon as it enters in `dev` I will add the support for vyos and arubacx

Thanks for the review. Merged. However, you might want to review the next PR first and then implement both.

**Pull request merged**